src/sdl/SDL.cpp

```diff
@@ -223,6 +223,15 @@ uint32_t screenMessageTime = 0;
 #define SOUND_ECHO 0.2
 #define SOUND_STEREO 0.15
 
+void freeSafe(void *ptr)
+{
+    if (ptr != NULL)
+    {
+	free(ptr);
+	ptr = NULL;
+    }
+}
+
```

Comment:
This is wrong for two reasons:

- you don't need to check for `NULL` when calling `free()`
- to set a pointer variable, you have to use a pointer to a pointer and pass the address of the pointer

So you could do something like this:

```cpp
void freeSafe(void** ptr)
{
    free(*ptr);
    *ptr = NULL;
}
...
...
    freeSafe(&some_ptr);

Reply:
Damm, I always took this for truth... I remember using this code for a long time. Oh well, I guess it is what it is.